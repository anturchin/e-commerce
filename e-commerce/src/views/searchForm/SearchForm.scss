.form-search {
    display: flex;
    flex-direction: row;
    align-items: center;

    gap: 20px;

    background: #222;
    z-index: 1000;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 40px;
    border-radius: 4px;
    box-shadow: 0 15px 35px rgba(0, 0, 0, 9);

    margin: 3rem 3rem;
}
