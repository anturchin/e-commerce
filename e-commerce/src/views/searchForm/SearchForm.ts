import { View } from '../View';
import { InputSearch } from './inputSearch/InputSearch';
import { ImgSearch } from './imgSearch/ImgSearch';
import { Button } from '../button/Button';

import './SearchForm.scss';
import { FormSearchElemntsType } from './types';

export class SearchForm extends View {
    private inputSearch: InputSearch | null = null;

    private buttonSubmit: Button | null = null;

    constructor() {
        super({ tag: 'form', classNames: ['form-search'] });
        this.setupForm();
    }

    public getFormElemnts(): FormSearchElemntsType | null {
        if (this.inputSearch && this.buttonSubmit) {
            return {
                searchInput: this.inputSearch,
                buttonSubmit: this.buttonSubmit,
            };
        }
        return null;
    }

    private setupForm(): void {
        this.inputSearch = new InputSearch({
            type: 'text',
            label: 'Search',
            required: true,
        });
        this.viewHtmlElement.addInnerElement(this.inputSearch.getElement());

        this.buttonSubmit = new Button({
            label: '',
        });

        const img = new ImgSearch().getElement();
        this.buttonSubmit.addInnerElement(img);
        this.viewHtmlElement.addInnerElement(this.buttonSubmit.getElement());
    }
}
