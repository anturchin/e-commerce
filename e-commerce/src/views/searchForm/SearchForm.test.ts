import { describe, expect, test } from '@jest/globals';
import { SearchForm } from './SearchForm';

describe('SearchForm', () => {
    test('should match', () => {
        const form = new SearchForm();
        const renderedSearchForm = form.getElement();
        expect(renderedSearchForm.outerHTML).toMatchSnapshot();
    });
});
