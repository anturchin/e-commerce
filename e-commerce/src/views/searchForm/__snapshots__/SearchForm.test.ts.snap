// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match 1`] = `"<form class="form-search"><input class="input" type="text" placeholder="Search" required="true"><button class="custom-button"><img class="search_img" src="png-mock"></button></form>"`;
